e2e/testcases/cli_test.go

```diff
@@ -1301,6 +1302,23 @@ func TestNomosMigrate(t *testing.T) {
 
 	nt.T.Cleanup(func() {
 		// Restore state of Config Sync installation after test
+		// Legacy ConfigManagement sets readiness on reconciler-manager and resource-group-controller,
+		// which isn't reliably cleaned up after tests. Delete both to ensure a fresh
+		// install instead of patching.
+		rmDeployment := k8sobjects.DeploymentObject(
+			core.Name(reconcilermanager.ManagerName),
+			core.Namespace(configsync.ControllerNamespace),
+		)
+		rgDeployment := k8sobjects.DeploymentObject(
+			core.Name(configmanagement.RGControllerName),
+			core.Namespace(configmanagement.RGControllerNamespace),
+		)
+		if err := nt.KubeClient.Delete(rmDeployment); err != nil && !apierrors.IsNotFound(err) {
+			nt.T.Error(err)
+		}
+		if err := nt.KubeClient.Delete(rgDeployment); err != nil && !apierrors.IsNotFound(err) {
+			nt.T.Error(err)
+		}
 		if err := nomostest.InstallConfigSync(nt); err != nil {
```

Comment:
Deleting and recreating specific subcomponents is going to be error prone, as it doesn't guarantee the e2e client claims ownership of all fields for all objects which could have drifted. Perhaps using client side apply here when reinstalling would be a more thorough option

Reply:
Mmm without the other PR it's hard to know whether further changes are effective or just false positive, some natural result from the standalone branching. I might need to add some check for the readiness field as well to verify it's actually removed.

Reply:
I'd suggest rebasing that PR on top of this one, then you can validate it works from the tests in the other PR